src/util/schema_keywords.js

```diff
@@ -0,0 +1,174 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const js_utils = require('./js_utils');
+const { _: CG } = require('ajv');
+
+/**
+ * @typedef {import('ajv').KeywordCxt} KeywordCxt
+ */
+
+const KEYWORDS = js_utils.deep_freeze({
+
+    methods: {
+        keyword: 'methods',
+    },
+
+    doc: {
+        keyword: 'doc',
+    },
+
+    // schema: { date: true } will match (new Date())
+    date: {
+        keyword: 'date',
+        errors: false,
+        // schemaType: 'boolean',
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * 
+         */
+        code(cxt) {
+            const d = cxt.it.data;
+            cxt.pass(CG` ${d} instanceof Date `);
+        }
+    },
+
+    // schema: { idate: true } will match (new Date()).getTime()
+    idate: {
+        keyword: 'idate',
+        // schemaType: 'boolean',
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * 
+         */
+        code(cxt) {
+            // TODO: Remove accepting instanceof Date after converting all uses of
+            // idates in the database schema into the date format (currently most of them are
+            // already saved in ISO format which is not idate)
+            // TODO: Remove accepting instanceof Date after converting all uses of
+            // idates in the database schema into the date format (currently most of them are
+            // already saved in ISO format which is not idate)
+            const d = cxt.it.data;
+            cxt.gen
+                .if(CG`
+                    Number.isInteger(${d}) &&
+                    !isNaN(new Date(${d}).getTime())
```

Comment:
I think this is clearer than !isNaN -
```suggestion
                    Number.isInteger(new Date(${d}).getTime())
```